_tutorials/faceted-search.md

```diff
@@ -0,0 +1,1845 @@
+          { "from": 5, "to": 25, "key": "5-25 miles away" },
+          { "from": 25, "key": "Over 25 miles away" }
+        ]
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains all three stores bucketed by distance:
+
+<details markdown="block">
+  <summary>
+    Response
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "took": 16,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 3,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": "stores",
+        "_id": "1",
+        "_score": 1,
+        "_source": {
+          "name": "Downtown Store",
+          "store_location": {
+            "lat": 40.751,
+            "lon": -73.99
+          }
+        }
+      },
+      {
+        "_index": "stores",
+        "_id": "2",
+        "_score": 1,
+        "_source": {
+          "name": "Suburban Store",
+          "store_location": {
+            "lat": 40.83,
+            "lon": -74.2
+          }
+        }
+      },
+      {
+        "_index": "stores",
+        "_id": "3",
+        "_score": 1,
+        "_source": {
+          "name": "Outskirts Store",
+          "store_location": {
+            "lat": 41.2,
+            "lon": -74.8
+          }
+        }
+      }
+    ]
+  },
+  "aggregations": {
+    "store_distance": {
+      "buckets": [
+        {
+          "key": "Within 5 miles",
+          "from": 0,
+          "to": 5,
+          "doc_count": 1
+        },
+        {
+          "key": "5-25 miles away",
+          "from": 5,
+          "to": 25,
+          "doc_count": 1
+        },
+        {
+          "key": "Over 25 miles away",
+          "from": 25,
+          "doc_count": 1
+        }
+      ]
+    }
+  }
+}
+```
+
+</details>
+
+## Advanced faceting scenarios
+
+The following sections provide examples of advanced faceted search functionality.
+
+### Post-filter faceting
+
+When a user selects the `red` color, you want to show all available colors for the base query (t-shirts) while also showing other facets (like sizes) filtered to only show options available for red products. Use `post_filter` to achieve this:
+
+```json
+POST /products/_search
+{
+  "query": {
+    "match": {
+      "name": "t-shirt"
+    }
+  },
+  "post_filter": {
+    "term": { "color": "red" }
+  },
+  "aggs": {
+    "all_colors": {
+      "terms": {
+        "field": "color"
+      }
+    },
+    "sizes_for_red": {
+      "filter": {
+        "term": { "color": "red" }
+      },
+      "aggs": {
+        "sizes": {
+          "terms": {
+            "field": "size"
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains the matching t-shirt and the color buckets for all t-shirts:
+
+<details markdown="block">
+  <summary>
+    Response
+  </summary>
+  {: .text-delta}
+
+```json
+{
```

Comment:
It would be better to show a image for the response. I try to ask AI to generate an ASCII diagram based on one response from here.

<img width="715" height="846" alt="Image" src="https://github.com/user-attachments/assets/47e91036-4202-40c0-9e46-7dc53384c941" />

You may not need to do this for every response. But just give one example case corresponding to one response.